File: 8_kyu/Count_the_number_of_cubes_with_paint_on.c
// Calculates x³.
static inline int pow3(const int x) {
  return x*x*x;
}

// Calculates outer cubes of a root-cube cutted 'cuts' times.
int countSquares(const int cuts) {
  // Argument out of range.
  if( cuts < 0 )
    return 0;
  // Calculate number of all cubes and the number of inner cubes
  const int total_cubes = pow3(cuts+1);
  const int inner_cubes = cuts < 2 ? 0 : pow3(cuts-1);
  // Return the number of cubes in the shell
  return total_cubes - inner_cubes;
}

File: 8_kyu/Count_the_number_of_cubes_with_paint_on.py
def count_squares(cuts :int) -> int:
    """
    Calculates outer cubes of a root-cube cutted 'cuts' times.
    """
    if cuts < 0:
        # argument out of range
        return 0
    # Calculate number of all cubes and the number of inner cubes
    total_cubes = (cuts+1)**3;
    inner_cubes = (cuts-1)**3 if cuts > 1 else 0
    # Return the number of cubes in the shell
    return total_cubes - inner_cubes;    
